src/contributing.md

```diff
@@ -188,8 +188,12 @@ happened to the copy in this repo and creates commits on the remote repo that ma
 changes. Every local commit that touched the subtree causes a commit on the remote repo, but is
 modified to move the files from the specified directory to the tool repo root.
 
+The target branch on the tool repo should be chosen carefully. 
 Make sure to not pick the `master` branch on the tool repo, so you can open a normal PR to the tool
-to merge that subrepo push.
+to merge that subrepo push. It is also recommended that you create 
+a brand new branch as the push target, however, 
+if you'd prefer to use an existing branch you will typically need to delete it first
+because changes cannot be fast forwarded.
```

Comment:
"changes cannot be fast forwarded" sounds like a quote of the git error message, but I don't think it's particularly informative - maybe say "subtree pushes rewrite history, so using an existing branch will require you to force push"?

Reply:
You can't force push these changes which is part of the problem and why existing branches can't be used.

I'm open to changing the text again if folks think it can be more clear, but I'm also trying to stick to the pattern of not explaining everything inline in the guide (which also skips crucial pieces of information mandatory for performing the syncs, e.g. patching git subtree itself), especially since I think the now-linked Clippy docs cover everything at that more detailed level.

From my perspective this section has historically just called out some of the major "gotchas", and the target branch is one of those "gotchas" that's been missing here. I'm hesitant to start trying to explain the various "whys" in more detail especially if doing so introduces ambiguity around the "what" actually needs to be conveyed.

Reply:
Hmm, ok, can you just rephrase to say "git will give an error that changes cannot ..." instead? I think that's more clear than just the phrase alone, especially since git's error isnt very good IMO.

Reply:
Do you feel like a statement to the effect of "do this because if you don't, git will give you this error" is all that helpful for users?

Again I'm open to changes, but here's the piece I'm struggling with... the existing material that's in the guide under https://rustc-dev-guide.rust-lang.org/contributing.html?highlight=subtree#synchronizing-a-subtree is not in a complete/sufficient state. Any user that attempts to do subtree sync via just the steps documented here will run into all sorts of issues and will never be able to perform the sync successfully.

Additionally, for the items that _are_ documented here, most directions/guidance don't drill into their respective why/rationale (e.g. no mention of why subtree vs. submodule, why a prefix is always needed on both push and pull, why "rebasing merge commits is bad in general", etc.). A complete description of the rationale/necessity for why a clean/new branch is needed on pushes would require a relatively lengthy and in-the-weeds description which wouldn't really match the surround content/level of detail. Similarly, trying to give a high level short summary doesn't seem to be working either/has the potential to introduce more confusion.

I'm starting to feel like the best path may be to cut out most of this section and just point over to the clippy docs which do cover the full, reproducible steps that have to be followed, with details around the "why" for edification, and various important tips and tricks that are useful when actually dealing with these activities.

Otherwise, I feel like this section would need to be greatly expanded to provide a holistic guide, and would be largely (if not entirely) duplicative of the existing clippy docs

Reply:
> I'm starting to feel like the best path may be to cut out most of this section and just point over to the clippy docs which do cover the full, reproducible steps that have to be followed, with details around the "why" for edification, and various important tips and tricks that are useful when actually dealing with these activities.

If you're willing to work on that, I think it would be a good idea :) otherwise I'm fine merging this as-is.

Reply:
👍 sounds like a plan

I can update this later today/tomorrow. I don't think there's an urgent need for the concept in the originally proposed text so I'm content with repurposing this PR for the above approach, though also fine opening a follow up if you want to go ahead and merge.

Reply:
Let's use this PR so you have less rebasing / git magic to worry about :)